src/gstreamer/gstkvssink.cpp

```diff
@@ -1188,6 +1188,23 @@ gst_kvs_sink_handle_sink_event (GstCollectPads *pads,
         }
         case GST_EVENT_EOS: {
             LOG_INFO("EOS Event received in sink for " << kvssink->stream_name);
+            
+            if(data && data->kinesis_video_stream) {
+                Frame eofr = EOFR_FRAME_INITIALIZER;
+                STATUS put_eofr_status = data->kinesis_video_stream->putFrame(eofr);
+                if(STATUS_FAILED(put_eofr_status)) {
+                    LOG_WARN("Failed to put EOFR for " << kvssink->stream_name);
+                }
+            } else {
+                LOG_WARN("NULL pointer! Failed to put EOFR for " << kvssink->stream_name);
```

Comment:
Saying null ptr in warn looks scary & looks like an error. This is handling the case where kinesis_video_stream failed to initialize?

Reply:
Yes and as a safety measure in case either of these two pointers get nulled. This correlates with a STATUS_NULL_ARG error we get from Producer for these kind of null checks. How does "Null argument, failed to put EOFR for " sound?

Reply:
I'm thinking that we should just make this an INFO log and say what it's doing literally.

"Kinesis Video Stream failed to initialize, not putting EOFR".

Reply:
I'm still leaning towards having it be a WARN since it 'failed' to initialize. @disa6302 any thoughts?